test/langtools/tools/javac/records/RecordCompilationTests.java

```diff
@@ -1289,16 +1273,15 @@ record R(int recordComponent) {}
         )) {
             File dir = assertOK(true, source);
             int numberOfFieldRefs = 0;
-            for (final File fileEntry : dir.listFiles()) {
+            for (final File fileEntry : Objects.requireNonNull(dir.listFiles())) {
                 if (fileEntry.getName().endsWith("R.class")) {
-                    ClassFile classFile = ClassFile.read(fileEntry);
-                    for (CPInfo cpInfo : classFile.constant_pool.entries()) {
-                        if (cpInfo instanceof ConstantPool.CONSTANT_Fieldref_info) {
+                    ClassModel classFile = Classfile.of().parse(fileEntry.toPath());
+                    for (int i = 1; i < classFile.constantPool().entryCount(); ++i) {
+                        if (classFile.constantPool().entryByIndex(i) instanceof FieldRefEntry fieldRefEntry) {
```

Comment:
I've identified iteration over CP entries is problematic in the Classfile API yet, see:
https://mail.openjdk.org/pipermail/classfile-api-dev/2023-August/000399.html

Here the int i must be incremented by the entry::width or the iteration will fail on double-slot entries (long and double).